src/random.cpp

```diff
@@ -192,20 +192,38 @@ uint64_t GetRdSeed() noexcept
 #elif defined(__aarch64__) && defined(HWCAP2_RNG)
 
 bool g_rndr_supported = false;
+bool g_rndrrs_supported = false;
+
+bool VerifyRNDRRS()
+{
+    uint8_t ok;
+    uint64_t test;
+    int max_retries = 10;
+    do {
+        __asm__ volatile("mrs %0, s3_3_c2_c4_1; cset %w1, ne;"
+                         : "=r"(test), "=r"(ok)::"cc");
+        if (ok) return true;
+        __asm__ volatile("yield");
+    } while (--max_retries > 0);
+    return false;
+}
```

Comment:
This function basically duplicates `GetRNDRRS()`; it should instead be refactored to avoid duplicating.

Reply:
Agreed, I'll refactor to avoid duplication with `GetRNDRRS()`.

Reply:
Thank you, I refactored, could you please take another look?

Reply:
I don't think there's any need to have these be separate commits, please squash.

Reply:
Squashed.